.buildkite/pipeline.yml

```diff
@@ -1,5 +1,44 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+env:
+  SETUP_GOLANG_VERSION: '1.18.10'
 
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+  - label: "Update"
+    key: update
+    command:
+      - ".buildkite/scripts/update.sh"
```

Comment:
It looks like that the Jenkins pipeline is intended to be run everything in the same agent
https://github.com/elastic/elastic-agent-client/blob/aa961fb53dddc0aadff1858582558967222eecb6/.ci/Jenkinsfile#L6

Should these steps be run in the same buildkite agent too ? 
Do lint or Test require what this steps generates (GRPC code) ?